assets/chef-recipes/cookbooks/mariadb/recipes/uninstall.rb

```diff
@@ -1,19 +1,49 @@
+service 'mariadb' do
+  action :stop
+end
+
+execute 'Reset iptables settings' do
+  command 'iptables -D INPUT -p tcp --dport 3306 -j ACCEPT -m state --state NEW'
+end
+
 case node[:platform_family]
-when "debian"
-  package "mariadb-common" do
+when 'debian'
+  package 'mariadb-common' do
+    action :purge
+  end
+  file '/etc/apt/sources.list.d/mariadb.list' do
```

Comment:
The resource apt_repository has `:remove` action that safely removes the repository description.